js/start.md

```diff
@@ -275,454 +229,570 @@ To install React-specific Amplify UI components, run the following command:
+{: .callout .callout--warning}
+
+## Step 3: Add API and Database
 
-Create the AWS backend resources:
+Add a GraphQL API to your app and automatically provision a database with the following command (accepting all defaults is OK):
 
 ```bash
-$ amplify push
+$ amplify add api     #select GraphQL, API Key
 ```
 
-You'll notice that the Amplify CLI is also creating an Auth resource for you (this is a dependency of the analytics resource).
+The `add api` flow above will ask you some questions, like if you already have an annotated GraphQL schema. If this is your first time using the CLI select **No** and let it guide you through the default project **"Single object with fields (e.g., “Todo” with ID, name, description)"** as it will be used in the code generation examples below. Later on you can always change it. This process creates an AWS AppSync API and connects it to an Amazon DynamoDB database.
 
-> After your analytics resource has been successfully created, the Amplify configuration file (`aws-exports.js`) will be added to the source directory. (The 'src' directory for your project is defined by your answer to the `Source Directory Path` question during `amplify init`.)
+[Learn more about annotating GraphQL schemas and data modeling](https://aws-amplify.github.io/docs/cli-toolchain/graphql){:target="_blank"}.
 
-**If You Change Your Mind**
+Create required backend resources for your configured api with the following command:
 
-You can update a service by running `amplify update <category-name>`. If you no longer want to use a service you can delete it with `amplify remove <category-name>`. Lastly, you can wipe out the whole project by running `amplify delete` (Warning: This will attempt to delete everything locally and in the cloud and reset your project as if you never ran `amplify init`).
+```bash
+$ amplify push
+```
 
-## Step 4. Integrate AWS Resources
+Since you added an API the `amplify push` process will automatically enter the codegen process and prompt you for configuration. Accept the defaults choosing JavaScript, TypeScript, or Angular depending on your platform. 
+<div id="react" class="tab-content">
+For this tutorial choose **JavaScript**.
+</div>
+<div id="react-native" class="tab-content">
+For this tutorial choose **JavaScript**.
+</div>
+<div id="purejs" class="tab-content">
+For this tutorial choose **JavaScript**.
+</div>
+<div id="vue" class="tab-content">
+For this tutorial choose **JavaScript**.
+</div>
+<div id="ionic" class="tab-content">
+For Ionic applications, choose **Angular** which will create an `API.service.ts` file in the app directory.
+</div>
+<div id="angular" class="tab-content">
+For Angular applications, choose **Angular** which will create an `API.service.ts` file in the app directory.
```

Comment:
You should probably put this between the add api and push steps.

Reply:
I didn't do that here because the codegen prompt is done during the push. Do you still think it should be moved?